cylc/flow/xtriggers/wall_clock.py

```diff
@@ -35,16 +36,23 @@ def wall_clock(offset=None, absolute_as_seconds=None, point_as_seconds=None):
         absolute_as_seconds (int):
             Satisfy this xtrigger after the specified time.
             Should be a datetime in the unix time format.
-        point_as_seconds (int):
-            Provided by Cylc. The cycle point in unix time format.
+        point (int):
+            Provided by Cylc. The ISO8601 cycle point.
 
     """
-    offset_as_seconds = 0
-    if offset is not None:
-        offset_as_seconds = int(interval_parse(offset).get_seconds())
     if absolute_as_seconds:
         trigger_time = absolute_as_seconds
     else:
-        trigger_time = point_as_seconds + offset_as_seconds
+        if not offset:
+            offset = 'PT0S'
+        trigger_time = int(
+            point_parse(
+                str(
+                    get_point_relative(
+                        offset, point
+                    )
+                )
+            ).get('seconds_since_unix_epoch')
```

Comment:
> is there a less convoluted way to do the offset computation?

I am not sure about the need to convert from metomi-isodatetime objects to Cylc date-time objects and back
https://github.com/cylc/cylc-flow/blob/d8d797b2ba42e5a5e1307e794d69721623db30ec/cylc/flow/cycling/iso8601.py#L866-L876

Can this not be handled by metomi-isodatetime alone, by adding the `Duration` to the `TimePoint`?

> this is probably quite inefficient compared to the old broken way? (complex ISO datetime computation in every clock check, instead of a simple float substraction). Can this be mitigated?

Unfortunately I think we must maintain the correct inexact units (years & months) if used. But I suppose an optimisation can be made if years or months are not in the offset: because all the other units are exact[^1], you could convert to seconds since unix epoch before doing the addition/subtraction in that case, might be faster.

[^1]: Not strictly speaking true for ISO 8601, but correct for our isodatetime implementation. See https://github.com/metomi/isodatetime/blob/99d509813bcee8c9fd64523221ef47d45a0c4a72/README.md?plain=1#L282-L294

Reply:
Fixed now, see updated description.